src/types/utils/EmptyObject.ts

```diff
@@ -0,0 +1,19 @@
+import Falsy from './Falsy';
+
+type EmptyObject = Record<string, never>;
+
+// eslint-disable-next-line rulesdir/no-negated-variables
+function isNotEmptyObject<T extends Record<string, unknown> | Falsy>(arg: T | EmptyObject): arg is T {
+    return Object.keys(arg ?? {}).length > 0;
+}
+
+function isEmptyObjectOrString<T extends Record<string, unknown> | Falsy>(arg: T | EmptyObject | Record<string, unknown> | string): arg is T {
+    if (typeof arg !== 'string' && typeof arg !== 'object') {
+        throw new TypeError('Argument must be a string or an object');
+    }
+
+    return Object.keys(arg ?? {}).length === 0;
+}
```

Comment:
```suggestion
function isEmptyObjectOrString<T extends Record<string, unknown> | string | Falsy>(arg: T): arg is T {
    if (arg && typeof arg !== 'string' && typeof arg !== 'object') {
      throw new TypeError('Argument must be a string or an object');
    }

    if (typeof arg === 'string') {
      return true;
    }

    return Object.keys(arg ?? {}).length === 0;
}
```

Let's simplify and fix the function to return true for strings:

[Check out the TS Playground.](https://www.typescriptlang.org/play?ts=5.2.2#code/C4TwDgpgBAYghgGwM4igXigVwHYBMIBmAlthLlAD5TaYIKVQGJIQDcAsAFBeiRQCiAWzCgA8gCMAVhADGwdFABKsgPYAnXAB4kwNSQDmAGmoQAbhDUA+Dt04EccoiuxQiSISJATpc0WoDKugaaACpQEAAewBB4SEqqGtpB2EZY2ADW2CoA7tiWDDp6KQzwyCCWABRwavoAXFAhAJT11fqucWEA3lxQva4EUFU1UABkI1C8ECoDrVAAhGgYAOSFBkuj45PTULMLyypSssBLjVDdnH19wAAWajkm2Q3gEPxqd2oVSwCCNZiCMfJBJgdFBxNA4FBVsV1DsXAcfMdGjZLgBfLg9PpEAYVLYzYaLZZQ-QnM4Yy5qCDATBqFy6TBsMlQNG2cmU6kubxHAB06QgICQQzaAH4hWcUY0uQgYvobuhFlAAAw2ZlcGTOJAqKWSlT6T40OhLYxuDxiQ6+ALJXX6hCNJGq9WaiDa3VLJjICCG9omrxm4B+QJFXVuli2mxq7AarUIHWfHD4YikXCe43CU0I-2WipxwgkMih+0Rx3Oz6dFHJ9ypn3pi2Biql-OccORp3Rl0AIjgbfL3s55oDBk+cBOdtsulQ516TaLrc+AEZu5Xe36awPZw2UVAZHBgDJrhUII0J5uHVGY0t58YD8r0Y2Ty2z50dvUO22mQvPEuM7XH3B6ksh0yoZAA)

Reply:
changed